README.developer.md

```diff
@@ -0,0 +1,18 @@
+# NMState Coding Guidelines
+
+- NMState is written primarily in Python, and its coding style should follow the best practices of Python coding unless otherwise declared.
+- PEP8 is holy.
+- Tests are holy. Production code must be covered by unit tests and basic integration tests.
+- Commit message should use the provided commit-template.txt as template:
+  You can set the template for nmstate by configuring git: `git config commit.template commit-template.txt` 
+- Packages, modules, functions, methods and variables should use underscore_separated_names.
+- Class names are in CamelCase.
+- Imports should be grouped in the following order:
+-- Standard library imports
+-- Related third party imports
+-- Local application-specific or library-specific imports.
+- All indentation is made of the space characters. Tabs are evil. In makefiles, however, tabs are obligatory.
+  White space between code stanzas are welcome. They help to create breathing while reading long code. However, splitting stanzas into helper functions could be even better.
+
```

Comment:
The preference of single ticks should be mentioned here and ideally also some tooling to help to conform to this (or we use double quotes for which tooling exists ;-)).

Reply:
I got used to single quotes and it feels more minimalistic.. the python way. As we are already there, and I like it so much, I will indeed add it :)
I do not have tooling, but from experience after you get used to it, it sticks.